### PR TITLE
Reintroduce 1404 by Inlead: Upgrade ding_place2book to new API.

### PR DESCRIPTION
**Pull request by inlead**
Revert "Merge pull request #523 from kasperg/1404-revert-take-2"

http://platform.dandigbib.org/issues/1404

### CODE REVIEW

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 0abf45f8b9f300bec9096761076f30a9c2266fdf
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 0abf45f8b9f300bec9096761076f30a9c2266fdf
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 0abf45f8b9f300bec9096761076f30a9c2266fdf
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 0abf45f8b9f300bec9096761076f30a9c2266fdf
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 0abf45f8b9f300bec9096761076f30a9c2266fdf
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

 * Commit: 6b54a932310411d629ed61822fc27efd4f473bb0
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 0abf45f8b9f300bec9096761076f30a9c2266fdf
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

 * Commit: 6b54a932310411d629ed61822fc27efd4f473bb0
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: bc86a4f039de45e21b142fe7c78176ea37532b10
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

 * Commit: 697f87d3e555197543ae7189b0ec4941a8043976
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: bc86a4f039de45e21b142fe7c78176ea37532b10
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

 * Commit: 697f87d3e555197543ae7189b0ec4941a8043976
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: a14408bc99d5bcb536a9d3620cfdfdcf39922da1
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: a14408bc99d5bcb536a9d3620cfdfdcf39922da1
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: a14408bc99d5bcb536a9d3620cfdfdcf39922da1
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

File: modules/ding_place2book/ding_place2book.module [hunk 1]
```diff
@@ -144,1046 +158,454 @@ function ding_place2book_permission() {
...
     }
-    // Check if sales period is in the future and ticket sale has not
-    // started yet.
-    elseif ($p2b_result->headers['sales-status'] == 'upcoming') {
-      // Check if waiting list is activated.
-      if ($p2b_result->headers['waiting-list'] == 'true') {
-        $p2b_ticket_result['type'] = 'order-waiting-list';
-        $p2b_ticket_result['url'] = $p2b_result->headers['location'];
+    elseif ($state === 'published') {
+      // If event is publihshed we need to look at ticket types and sale
```
Comment (JrgenGNielsen) [inline modules/ding_place2book/ding_place2book.module]: typo

File: modules/ding_place2book/ding_place2book.admin.inc [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+/**
+ * Load and parse XML with export category.
+ *
+ * @return array
+ *   Array with export category.
+ *
+ * @see http://www.kultunaut.dk/perl/export/nautgenrerxml
+ */
+function _ding_place2book_load_export_category() {
+  $path = 'http://www.kultunaut.dk/perl/export/nautgenrerxml';
```
[the remark this answers is not in the document]
Comment (holt83) [reply modules/ding_place2book/ding_place2book.admin.inc]: I have asked the business. Will report back here.

File: modules/ding_place2book/ding_place2book.info [hunk 3]
```diff
@@ -2,9 +2,18 @@ name = Ding Place2book
...
 dependencies[] = ding_event
+dependencies[] = features
 dependencies[] = number
-dependencies[] = ctools
 dependencies[] = panels
 dependencies[] = page_manager
+dependencies[] = strongarm
+dependencies[] = xautoload
 configure = admin/config/ding/place2book
+explanation = Field type(s) in use - see <a href="/admin/reports/fields">Field list</a>
```
[the remark this answers is not in the document]
Comment (holt83) [reply modules/ding_place2book/ding_place2book.info]: I was actually wondering about this too. @guddo

[the remark this answers is not in the document]
Comment (guddo) [reply modules/ding_place2book/ding_place2book.info]: I don't, however it's done by fields_ui module. Should we just remove it?

File: modules/ding_place2book/ding_place2book.admin.inc [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+    '#default_value' => isset($settings['token']) ? $settings['token'] : '',
+    '#description' => t('This is the default API key used when connecting to Place2book.'),
+  );
+
+  $form['ding_place2book']['paths']['#tree'] = TRUE;
+
+  $form['ding_place2book']['paths']['eventMakers'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Insert Place2Book event makers path without leading slash.'),
+    '#default_value' => isset($settings['eventMakers']) ? substr($settings['eventMakers'], $base_length) : 'event_makers',
```
[the remark this answers is not in the document]
Comment (holt83) [reply modules/ding_place2book/ding_place2book.admin.inc]: Fixed

File: modules/ding_place2book/ding_place2book.admin.inc [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+ * Submit handler for configuration page of p2b library.
+ */
+function ding_place2book_service_admin_form_submit($form, &$form_state) {
+  $values = $form_state['values'];
+  $settings = $values['ding_place2book'];
+  $base_url = $settings['url'];
+
+  // Convert from paths to absolute URLs by adding the base URL.
+  foreach ($settings['paths'] as $key => $path) {
+    $settings[$key] = $base_url . '/' . $path;
```
[the remark this answers is not in the document]
Comment (holt83) [reply modules/ding_place2book/ding_place2book.admin.inc]: Fixed

File: modules/ding_place2book/ding_place2book.module (hunk 1 above)
Comment (holt83) [reply modules/ding_place2book/ding_place2book.module]: Fixed

File: modules/ding_place2book/ding_place2book.admin.inc (hunk 2 above)
[the remark this answers is not in the document]
Comment (holt83) [reply modules/ding_place2book/ding_place2book.admin.inc]: I talked with Kultunaut. It's very rare that they change the URL and if they do, they are aware that we are dependent on it and they will contact DDB in advance.

So I think it's not worth the trouble and we should just change the URL in code if needed.

**Comment by JrgenGNielsen** [comment]
reviewed and approved

**Comment by JrgenGNielsen** [comment]
reviewed and approved

**Pull request merged**